rig.js

```diff
@@ -127,6 +127,30 @@ class RigManager {
     this.localRig.textMesh.sync();
   }
 
+  setLocalAvatarImage(avatarImage) {
+    const geometry = new THREE.CircleBufferGeometry(0.1, 32);
+    const img = new Image();
+    img.src = avatarImage;
+    img.crossOrigin = 'Anonymous';
+    img.onload = () => {
+      texture.needsUpdate = true;
+    };
+    img.onerror = err => {
+      console.warn(err.stack);
+    };
+    const texture = new THREE.Texture(img);
+    const material = new THREE.MeshBasicMaterial({
+      map: texture,
+      side: THREE.DoubleSide,
+    });
+    const avatarMesh = new THREE.Mesh(geometry, material);
+    avatarMesh.position.x = -0.5;
+    avatarMesh.position.y = -0.02;
+    this.localRig.textMesh.add(avatarMesh);
```

Comment:
I suspect this will "paper over" the existing avatar mesh without removing it, and additional sets will further add more icons.

https://github.com/webaverse/app/blob/fbb507c4d0e2cc10cf3ef2251dabacf084c34b54/rig.js#L39

It probably makes sense to remove that icon rather than add ones on top.

Reply:
1a301bc